src/Fractions/Fraction.cs

```diff
@@ -146,4 +153,180 @@ namespace Fractions;
+    /// <remarks>
+    ///     A Fraction is considered to be in canonical form if its denominator is one,
+    ///     or if its numerator and denominator are coprime numbers (their greatest common divisor is one).
+    /// </remarks>
+    public static bool IsCanonical(Fraction value) {
+        var numerator = value.Numerator;
+        var denominator = value.Denominator;
+        if (denominator.IsOne) {
+            return true;
+        }
+
+        if (numerator.IsZero) {
+            return denominator.IsZero; // if we want to consider NaN as "canonical"
+        }
+
+        if (denominator.IsZero) {
+            return numerator.IsOne || numerator == BigInteger.MinusOne;
+        }
+
+        return BigInteger.GreatestCommonDivisor(numerator, denominator).IsOne;
+    }
+
+    /// <summary>Determines if a value represents an even integral number.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is an even integer; otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool IsEvenInteger(Fraction value) {
+        var fraction = value.Reduce();
+        return fraction.Denominator.IsOne && fraction.Numerator.IsEven;
+    }
+
+    /// <summary>Determines if a value is finite.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is finite; otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool IsFinite(Fraction value) {
+        return !value.Denominator.IsZero;
+    }
+
+    /// <summary>Determines if a value represents an integral number.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is an integer; otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool IsInteger(Fraction value) {
+        var denominator = BigInteger.Abs(value.Denominator);
+        if (denominator.IsOne) {
+            return true;
+        }
+
+        if (denominator.IsZero) {
+            return false;
+        }
+
+        var numerator = BigInteger.Abs(value.Numerator);
+        if (numerator.IsZero) {
+            return true;
+        }
+        
+        return numerator >= denominator && (numerator % denominator).IsZero;
+    }
+
+    /// <summary>Determines if a value represents an odd integral number.</summary>
+    /// <param name="value">The value to be checked.</param>
+    /// <returns>
+    ///     <see langword="true" /> if <paramref name="value" /> is an odd integer; otherwise, <see langword="false" />.
+    /// </returns>
+    public static bool IsOddInteger(Fraction value) {
+        var fraction = value.Reduce();
+        return fraction.Denominator.IsOne && !fraction.Numerator.IsEven;

```

Comment:
Can we optimize the method a little bit with this?

```suggestion
        if (value.Denominator.IsZero) {
            return false;
        }
        if (!value._normalizationNotApplied) {
            return BigInteger.Abs(value.Denominator).IsOne && !value.Numerator.IsEven;
        }
        var intValue = BigInteger.DivRem(value.Numerator, value.Denominator, out var remainder);
        return remainder.IsZero && !intValue.IsEven;
```

Reply:
fixed in https://github.com/Fractions-DotNet/Fractions/pull/96/commits/1643913092e03ca145a91e4d5de95948f16949da